### PR TITLE
adding nexttoken and maxResults for snapshot

### PR DESCRIPTION
**Pull request by rjain-iitr**

### CODE REVIEW

File: ec2api/api/snapshot.py [hunk 1]
```diff
@@ -89,6 +95,29 @@ def get_os_items(self, ids, max_results, next_token, detail):
...
+        
+        next_item = 0 
+        if next_token:
+           for i, elem in enumerate(formatted_items):
+              if next_token == elem[SORT_KEY]:
+                 next_item = i+1
+        if next_item:
+            formatted_items = formatted_items[next_item:]
+        if max_results and max_results < len(formatted_items):
+            next_token = base64.b64encode(str(next_item + max_results))
```
Comment (aggarwalchirag) [inline ec2api/api/snapshot.py]: Why do we need to calculate next_token here ?

File: ec2api/api/snapshot.py [hunk 2]
```diff
@@ -89,6 +95,29 @@ def get_os_items(self, ids, max_results, next_token, detail):
...
+        if max_results and max_results > 1000:
+            max_results = 1000
+        formatted_items = sorted(formatted_items,
+                                 key=operator.itemgetter(SORT_KEY))
+        
+        next_item = 0 
+        if next_token:
+           for i, elem in enumerate(formatted_items):
+              if next_token == elem[SORT_KEY]:
+                 next_item = i+1
```
Comment (aggarwalchirag) [inline ec2api/api/snapshot.py]: What if someone calls up the API with the id of the last item as next_token.
formatted_items[next_item:] will give out-of-index error

File: ec2api/api/snapshot.py [hunk 3]
```diff
@@ -137,6 +167,6 @@ def _format_snapshot(context, os_snapshot):
             'description': os_snapshot.description,
             'snapshotId': os_snapshot.id,
             'volumeId': os_snapshot.volume_id,
-            'volumeSize': os_snapshot.size,
+            'snapshotSize': os_snapshot.size,
```
Comment (aggarwalchirag) [inline ec2api/api/snapshot.py]: We want to keep it as volumeSize only. AWS also displays volumeSize

Comment (rjain-iitr) [inline ec2api/api/snapshot.py]: ok will revert this

File: ec2api/api/snapshot.py (hunk 2 above)
Comment (rjain-iitr) [inline ec2api/api/snapshot.py]: It doesnot give out-of-index error for last next_token

File: ec2api/api/snapshot.py (hunk 1 above)
Comment (rjain-iitr) [inline ec2api/api/snapshot.py]: removing it. We donot need this

File: ec2api/api/snapshot.py [hunk 4]
```diff
@@ -89,6 +95,31 @@ def get_os_items(self, ids, max_results, next_token, detail):
             return [clients.cinder(self.context).backups.get(ids)]
 
 
+def get_paged(self, formatted_items, max_results, next_token):
+        SORT_KEY = 'snapshotId'
+        if not max_results and not next_token:
+            return formatted_items
+
+        if max_results and max_results > 1000:
```
Comment (aggarwalchirag) [inline ec2api/api/snapshot.py]: what is the purpose of comparing max_results with 1000

Comment (rjain-iitr) [inline ec2api/api/snapshot.py]: If the results are 1000 also we are getting 1000 results only before calling this function.

**Pull request merged**